app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java

```diff
@@ -44,54 +44,21 @@ public SearchResultHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(final SearchResultHolder holder, int position) {
-        Datum datum = datumList.get(position);
+        final Datum datum = datumList.get(position);
+        holder.previewImageView.setVisibility(View.GONE);
         if (datum != null) {
             holder.titleTextView.setText(Html.fromHtml(datum.getTitle()));
             holder.descriptionTextView.setText(Html.fromHtml(datum.getDescription()));
-            if (!TextUtils.isEmpty(datum.getLink())) {
-                LinkPreviewCallback linkPreviewCallback = new LinkPreviewCallback() {
-                    @Override
-                    public void onPre() {
-                        holder.previewImageView.setVisibility(View.GONE);
-                        holder.descriptionTextView.setVisibility(View.GONE);
-                        holder.titleTextView.setVisibility(View.GONE);
-                        holder.previewLayout.setVisibility(View.GONE);
+            holder.previewLayout.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Uri webpage = Uri.parse(datum.getLink());
+                    Intent intent = new Intent(Intent.ACTION_VIEW, webpage);
+                    if (intent.resolveActivity(context.getPackageManager()) != null) {
+                        context.startActivity(intent);
                     }
-
-                    @Override
-                    public void onPos(final SourceContent sourceContent, boolean b) {
-                        holder.previewLayout.setVisibility(View.VISIBLE);
-                        holder.previewImageView.setVisibility(View.VISIBLE);
-                        holder.descriptionTextView.setVisibility(View.VISIBLE);
-                        holder.titleTextView.setVisibility(View.VISIBLE);
-                        holder.titleTextView.setText(sourceContent.getTitle());
```

Comment:
There might be some small error or something but that doesn't mean you remove the complete code of it.